### DISCUSSION TITLE
taro 3.4.11 Uncaught ReferenceError: ENABLE_INNER_HTML is not defined

### DISCUSSION THREAD

**Discussion by ApeWhoLovesCode** [NervJS/taro#13248] [Q & A] [answered]
### 相关平台
H5

### 复现仓库
[git@github.com:ApeWhoLovesCode/test-react-taro.git](git@github.com:ApeWhoLovesCode/test-react-taro.git)
**浏览器版本: Chrome 109**
**使用框架: React**

### 复现步骤
1. create-react-app 创建的项目
2. import Taro from '@tarojs/taro';
3. 只要Taro调用方法就报错：ENABLE_INNER_HTML is not defined

```
function App() {
  return (
    <div className="App">
      <Button onClick={() => {
        Taro.showToast({title: '啊啊啊'})
      }}>Taro button</Button>
    </div>
  );
}
```

### 期望结果
能正确调用 `Taro` 的方法

### 实际结果
调用 `Taro` 方法就报错：Uncaught ReferenceError: ENABLE_INNER_HTML is not defined

### 环境信息
```
👽 Taro v3.4.11


  Taro CLI 3.4.11 environment info:
    System:
      OS: macOS 12.3.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.14.2 - /usr/local/bin/node
      Yarn: 1.22.19 - ~/.yarn/bin/yarn
      npm: 8.5.0 - /usr/local/bin/npm
    npmPackages:
      @tarojs/components: 3.4.11 => 3.4.11 
      @tarojs/mini-runner: 3.4.11 => 3.4.11 
      @tarojs/plugin-html: 3.4.11 => 3.4.11 
      @tarojs/react: 3.4.11 => 3.4.11 
      @tarojs/runtime: 3.4.11 => 3.4.11 
      @tarojs/taro: 3.4.11 => 3.4.11 
      @tarojs/webpack-runner: 3.4.11 => 3.4.11 
      babel-preset-taro: 3.4.11 => 3.4.11 
      eslint-config-taro: 3.4.11 => 3.4.11 
      react: ^17.0.0 => 17.0.2 
    npmGlobalPackages:
      typescript: 4.9.5
```

---
### 补充信息
我看了其他的issues，都是说配置jest.config，但是我配置了还是无效。
复现demo: https://github.com/ApeWhoLovesCode/test-react-taro

**Comment by TheKonka** [comment]
https://github.com/NervJS/taro/issues/8821
https://nervjs.github.io/taro-docs/docs/next/external-libraries
别人配置 `jest.config` 是因为`jest`测试报错吧
加上 `new webpack.DefinePlugin` 就不报错了
同时 引入也需要改下
```diff
- import Taro from '@tarojs/taro';
+ import { showToast } from '@tarojs/taro';
```

**Comment by ApeWhoLovesCode** [comment]
> #8821 https://nervjs.github.io/taro-docs/docs/next/external-libraries 别人配置 `jest.config` 是因为`jest`测试报错吧 加上 `new webpack.DefinePlugin` 就不报错了 同时 引入也需要改下
> 
> ```diff
> - import Taro from '@tarojs/taro';
> + import { showToast } from '@tarojs/taro';
> ```

按那里的评论 new webpack.DefinePlugin 配置完，还有您提的修改都尝试了，还是报一样的错。

**Comment by TheKonka** [comment]
![image](https://user-images.githubusercontent.com/22173084/217193641-21a1407b-6221-4218-8147-72ed18bfb997.png)
我试了可以了，评论里的不全，你参照 https://nervjs.github.io/taro-docs/docs/next/external-libraries/ 这个看看

**Comment by ApeWhoLovesCode** [comment]
> 

尝试了下，出现其他错误了，我觉得是我配置错地方了，您能提个合并，我看看改的哪里吗😂

**Comment by TheKonka** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/22173084/217203545-cba4dabb-0c72-45ff-b50b-19590a994dac.png)

![image](https://user-images.githubusercontent.com/22173084/217203483-9f1b6076-76ad-4f04-97e5-7e11b1ca0903.png)

**Comment by ApeWhoLovesCode** [comment]
> ![image](https://user-images.githubusercontent.com/22173084/217203545-cba4dabb-0c72-45ff-b50b-19590a994dac.png)
> 
> ![image](https://user-images.githubusercontent.com/22173084/217203483-9f1b6076-76ad-4f04-97e5-7e11b1ca0903.png)

可以了，感谢大佬！！~\(≥▽≤)/~
